### DISCUSSION TITLE
Using "HTMarch.dll" in Python on Linux

### DISCUSSION THREAD

**Discussion by m-shakeel-m** [OpenHantek/OpenHantek6022#336] [Q&A] [answered]
Hello, hope you all are doing fine.

I am using HTMarch dynamic link library in a Python (32-bit) code on Windows as
`dll = ctypes.WinDLL("....../HTMarch.dll")`
It works fine and I can easily access the functions defined in this library like
`dll.dsoGetCalLevel(m_nDevIndex, m_nCalData, 32)`, `dll.dsoSetVoltDIV(m_nDevIndex, 0, m_nCH1VoltDIV)` etc.

Now, I wanted to run the same Python code on Ubuntu and as you know it does not support DLL rather it requires SO file. I wanted to know how can I create this SO file or is there any other method that I can use in order to get this code working?

I have tried to use the VCprj folder given in Hantek's SDK folder in Visual Studio project but don't know how to generate SO file from that SDK.

**Comment by Ho-Ro** [comment, marked as answer]
Muhammad,
OpenHantek does not use Hantek's closed-source firmware or their Windows libraries. It uses its own firmware along with Python bindings to access scope functions from Python. Please see my repo [Hantek6022API](https://github.com/Ho-Ro/Hantek6022API). There are also a lot of Python scripts showing usage in the `examples` directory.
HTH, Martin

**Reply by m-shakeel-m** [reply]
Thanks a lot man! This API is a lifesaver.
